modules/reactor/include/meshgenerators/ConcentricCircleGeneratorBase.h

```diff
@@ -0,0 +1,96 @@
+  /// Number of rings in each circle or in the enclosing square
+  const std::vector<unsigned int> _ring_intervals;
+  /// Bias values used to induce biasing to radial meshing in ring regions
+  const std::vector<Real> _ring_radial_biases;
+  /// Widths, fractions, radial sectors and growth factors of the inner boundary layers of the ring regions
+  multiBdryLayerParams _ring_inner_boundary_layer_params;
+  /// Widths, fractions, radial sectors and growth factors of the outer boundary layers of the ring regions
+  multiBdryLayerParams _ring_outer_boundary_layer_params;
+  /// Subdomain IDs of the ring regions
+  std::vector<subdomain_id_type> _ring_block_ids;
+  /// Subdomain Names of the ting regions
+  std::vector<SubdomainName> _ring_block_names;
+  /// Volume preserving function is optional
+  const bool _preserve_volumes;
+  /// Shift in default subdomain IDs to avert potential conflicts
+  const subdomain_id_type _block_id_shift;
+  /// Whether inward interface boundaries are created
+  const bool _create_inward_interface_boundaries;
+  /// Whether outward interface boundaries are created
+  const bool _create_outward_interface_boundaries;
+  /// Shift in default boundary IDs of interfaces to avert potential conflicts
+  const boundary_id_type _interface_boundary_id_shift;
+  /// Whether the side-specific external boundaries are generated or not
+  const bool _generate_side_specific_boundaries;
+  /// Boundary ID of the mesh's external boundary
+  const boundary_id_type _external_boundary_id;
+  /// Boundary Name of the mesh's external boundary
+  const std::string _external_boundary_name;
+  /// Boundary Names of the mesh's inward interface boundaries
+  const std::vector<std::string> _inward_interface_boundary_names;
+  /// Boundary Names of the mesh's outward interface boundaries
+  const std::vector<std::string> _outward_interface_boundary_names;
+
+  /**
+   * Assign interface boundary names to the mesh if applicable.
+   * @param mesh Mesh to which the interface boundary names are assigned
+   */
+  void assignInterfaceBoundaryNames(ReplicatedMesh & mesh);
+
+  /**
+   * Assign block IDs and names to the mesh if applicable.
+   * @param mesh Mesh to which the block IDs and names are assigned
+   * @param block_ids_old old block ids that are assigned by default
+   * @param block_ids_new new block ids that are user-defined
+   * @param block_names block names that are user-defined
+   * @param generator_name class name of the mesh generator
+   */
+  void assignBlockIdsNames(ReplicatedMesh & mesh,
+                           std::vector<subdomain_id_type> & block_ids_old,
+                           std::vector<subdomain_id_type> & block_ids_new,
+                           std::vector<SubdomainName> & block_names,
+                           const std::string & generator_name);
+
+  /**
+   * Prepare user-defined ring block IDs and names to replace the default ones.
+   * @param block_it block iterator
```

Comment:
```suggestion
   * @param current_block_id last block id used in the ring block ids
```

Reply:
It's actually a counter. Will clarify it.